### DISCUSSION TITLE
Split PR into two?

### DISCUSSION THREAD

**Discussion by broksonic21** [renovatebot/renovate#8731] [Suggest an Idea]
Let’s say we are configured to group our npm package updates one way.

Everything is going fine, but one day a PR comes in with 4 package updates, which is normal. But I look at the PR and notice 2 are a bit concerning, so I want to hold them back for later to review when I get to a computer.

Is there a way to tell the PR to “split itself” into two just for this one time? So I can end up with one PR for the two packages I’m good to merge right now, and a second PR for the remaining packages? (Or temporarily tell renovate to remove the 2 held packages from the PR until it rebases again or is otherwise recreated, so I can merge just the 2 items good to go?) 

I’m not looking to permanently change the grouping, but I’ve run into this need quite it the a few times so hoping there’s a solve.

My ideal world is something easy to do from the phone (as opposed to from computer), as I am often reviewing PRs from phone while in transit.

Thanks!

**Comment by rarkins** [comment]
Unfortunately no, there is no ability to do this today. Although it sounds simple, there's quite a lot of conditions to think about. e.g. what exactly does "temporary" mean (until the PR is merged?), what happens when newer versions come out, etc.

Maybe PRs could be populated with a list of checkboxes - one for each package included - and then you tick the checkbox for any you want removed?

I'm happy for this type of idea and if we can work out a feasible way of doing it then that would be great, but my first impression is that it would be really complicated under the hood.

**Reply by broksonic21** [reply]
I'd love the checkbox idea to "remove" idea. I was thinking that would persist until the merge (or close) of the current PR. It would show up again if the PR was recreated for any reason (rebase, new schedule, merge this one and thus a new PR is generated, etc)

i.e. just a simple way to turn a _specific_ PR into a smaller set for purposes of merging.

**Comment by bo0tzz** [comment]
I just found myself desiring this idea for checkboxes to split a PR. This thread is pretty old by now, @rarkins what are your thoughts on this feature these days? With some pointers I might be willing to try my hand at this, if you think it's feasible.

**Reply by jamietanna** [reply]
Rhys isn't maintaining the project any more - but I'll take a look and get back to you this week

Immediate thoughts:

- it'd still likely be "too difficult" as things currently stand
- if we had 5 package updates to split, how would we indicate that? Would there need to be checkboxes for all 5 packages?
  - What if it's a much larger PR?
- what if you want to split out by `packageFileDir` on the repository, rather than the package itself?

**Reply by bo0tzz** [reply]
Ah I wasn't aware, sorry for the spurious ping!
> Would there need to be checkboxes for all 5 packages?

That was my thought yes. Basically adding another column to the table; here's a mockup: [^1]

| Package | Type | Update | Change | Pending | Split |
|---|---|---|---|---|---|
| [actions/setup-node](https://redirect.github.com/actions/setup-node) | action | major | `v5.0.0` -> `v6.0.0` |  | <ul><li>- [ ] </li></ul> |
| [astral-sh/setup-uv](https://redirect.github.com/astral-sh/setup-uv) | action | major | `v6.8.0` -> `v7.1.0` | `v7.1.1` | <ul><li>- [ ] </li></ul> |
| [github/codeql-action](https://redirect.github.com/github/codeql-action) | action | major | `v3.30.8` -> `v4.30.8` | `v4.30.9` | <ul><li>- [ ] </li></ul> |
| [immich-app/devtools](https://redirect.github.com/immich-app/devtools) | action | major | `use-mise-action-0.0.1` -> `use-mise-action-v1.1.0` |  | <ul><li>- [ ] </li></ul> |
| [immich-app/devtools](https://redirect.github.com/immich-app/devtools) | action | major | `multi-runner-build-workflow-v0.2.1` -> `multi-runner-build-workflow-v1.0.0` |  | <ul><li>- [x] </li></ul> |

> what if you want to split out by packageFileDir

I'm not sure I quite understand this use case, but my hunch is that should maybe be a rule in the config rather than an ad-hoc action?

[^1]: nb: it seems like the usual `- [ ]` syntax doesn't work in a table. Instead I had to use `<ul><li>- [ ] </li></ul>`.
